iroha_macro/iroha_derive/src/lib.rs

```diff
@@ -268,7 +294,17 @@ fn impl_from_variant(ast: &syn::DeriveInput) -> TokenStream {
                     .first()
                     .expect("Won't fail as we have more than  one argument for variant")
                     .ty;
-                return Some(from_variant(name, &variant.ident, variant_type));
+                let try_into = try_into_variant(name, &variant.ident, variant_type);
+                let from = if attrs_has_ident(&unnamed.unnamed[0].attrs, SKIP_FROM_ATTR) {
+                    quote!()
+                } else {
+                    from_variant(name, &variant.ident, variant_type)
+                };
```

Comment:
I think it is better to separate TryIntoVariant and FromVariant into separate derive macros.

Reply:
They are semantically similar and it is handy to have both `From<Variant> for Enum` and `TryFrom<Enum> for Variant`.

Reply:
Yes, but it is confusing when you add `FromVariant` macro and now you also have `TryIntoVariant` impls.

Reply:
Also TryInto is only used in a few places, while `From` is used in most of them.

Reply:
Can't agree with you here. It is not the same as `Clone` and `Copy` derives. This 2 derives fall into same path of conversion for enums. In that sense it would be nice to have both `impl` in same derive.

It certainly won't break codebase and will add extra possibilities to enums.

Reply:
Ok, let's leave it as 1 derive then.